lib/Frontend/Frontend.cpp

```diff
@@ -507,12 +513,13 @@ bool CompilerInstance::setUpModuleLoaders() {
     Context->addModuleLoader(std::move(PIML), false, false, true);
   }
 
-  std::unique_ptr<SerializedModuleLoader> SML =
-    SerializedModuleLoader::create(*Context, getDependencyTracker(), MLM,
+  if (!ImplicitModulesDisabled) {
```

Comment:
Can we put this to the else branch of the previous if? It ensures one of these `SerializedModuleLoaderBase` are instantiated.

Reply:
@nkcsgexi , I had to undo this change because it caused test-case failures. The order in which these loaders get added matters and this was moving the SerializedModuleLoader ahead of the ModuleInterfaceLoader and caused problems.